test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,340 @@
+    return;
+  }
+  if (expect_client_create) {
+    expectClientCreate(0);
+  }
+  expectStreamCreate(0);
+  ENVOY_LOG_MISC(trace, "Triggered interval timer");
+  test_sessions_[0]->interval_timer_->invokeCallback();
+}
+
+void HttpHealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+  // Timeout timer needs to be explicitly enabled, usually by a call to onIntervalBase().
+  if (!test_sessions_[0]->timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping trigger timeout timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered timeout timer");
+  test_sessions_[0]->timeout_timer_->invokeCallback(); // This closes the client, turns off timeout
+                                                       // and enables interval
+  if (!last_action) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from network timeout");
+    triggerIntervalTimer(true);
+  }
+}
+
+void HttpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type, bool last_action) {
+  test_sessions_[0]->client_connection_->raiseEvent(event_type);
+  if (!last_action && event_type != Network::ConnectionEvent::Connected) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream from close event");
+    triggerIntervalTimer(
+        true); // Interval timer is guaranteed to be enabled from a close event - calls
+               // onResetStream which handles failure, turning interval timer on and timeout off
+  }
+}
+
+void TcpHealthCheckFuzz::allocTcpHealthCheckerFromProto(
+    const envoy::config::core::v3::HealthCheck& config) {
+  health_checker_ = std::make_shared<TcpHealthCheckerImpl>(
+      *cluster_, config, dispatcher_, runtime_, random_,
+      HealthCheckEventLoggerPtr(event_logger_storage_.release()));
+  ENVOY_LOG_MISC(trace, "Created Tcp Health Checker");
+}
+
+void TcpHealthCheckFuzz::initialize(test::common::upstream::HealthCheckTestCase input) {
+  allocTcpHealthCheckerFromProto(input.health_check_config());
+  cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+      makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectClientCreate();
+  health_checker_->start();
+  if (input.health_check_config().has_reuse_connection()) {
+    reuse_connection_ = input.health_check_config().reuse_connection().value();
+  }
+
+  if (DurationUtil::durationToMilliseconds(input.health_check_config().initial_jitter()) != 0) {
+    interval_timer_->invokeCallback();
+  }
+}
+
+void TcpHealthCheckFuzz::respond(std::string data, bool last_action) {
+  if (!timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping response.");
+    return;
+  }
+  Buffer::OwnedImpl response;
+  response.add(data);
+
+  ENVOY_LOG_MISC(trace, "Responded with {}. Length (in bytes) = {}. This is the string passed in.",
+                 data, data.length());
+  read_filter_->onData(response, true);
```

Comment:
I honestly don't know. What does this affect?

Reply:
I think this is a carry over from http where http I didn't implement partial responses, because body and trailers didnt do anything, but honestly I think I'm kind of blanking on this.

Reply:
Looking into this myself, tcp drops the boolean so it doesn't matter 
https://github.com/envoyproxy/envoy/blob/fe17c1c95e26fbd34266b7a8c1f1acd7d78ba41a/source/common/upstream/health_checker_impl.h#L242